resolwe_bio/processes/import_data/seq_reads.yml

```diff
@@ -218,3 +218,220 @@
       re-save-file-list fastqc_archive2 "${RW_FASTQC[@]}"
       re-save-list fastqc_url "${FW_FASTQC_URL[@]}"
       re-save-list fastqc_url2 "${RW_FASTQC_URL[@]}"
+
+- slug: basespace-fastq-single
+  name: BaseSpace Reads (single-end)
+  requirements:
+    expression-engine: jinja
+    executor:
+      docker:
+        image: resolwebio/rnaseq
+  data_name: 'BaseSpace FASTQ single-end {{ reads.files.0.file }}'
+  version: 0.0.1
+  type: data:reads:fastq:single
+  flow_collection: sample
+  category: upload
+  persistence: RAW
+  description: >
+    Import NGS reads in FASTQ format from BaseSpace.
+  input:
+    - name: reads
+      label: BaseSpace NGS reads (FASTQ) files
+      type: data:basespace
+  output:
+    - name: fastq
+      label: Reads file
+      type: list:basic:file
+    - name: fastqc_url
+      label: Quality control with FastQC
+      type: list:basic:file:html
+    - name: fastqc_archive
+      label: Download FastQC archive
+      type: list:basic:file
+  run:
+    runtime: polyglot
+    language: bash
+    program: |
+      {% for read in reads.files %}
+        GZ_FILE_PATH={{read.file}}
+        cp ${GZ_FILE_PATH} .
+      {% endfor %}
+
+      gzip -dk *.fastq.gz
```

Comment:
No need to extract reads, `FastQC` [works with compressed files](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/2%20Basic%20Operations/2.1%20Opening%20a%20sequence%20file.html) as well.

Reply:
Is the `--extract ` in `fastqc "${FASTQGZ[@]}" --extract` necessary if gzipped files are on the input?

Reply:
Yes. The `--extract` flag is not related to the type of input but controls the output of the `FastQC` tool. From the [documentation](https://wiki.gacrc.uga.edu/wiki/FastQC):
```
    --extract       If set then the zipped output file will be uncompressed in
                    the same directory after it has been created.  By default
                    this option will be set if fastqc is run in non-interactive
                    mode.
```

Reply:
Since we're running `FastQC` in non-interactive mode, it seems that the `--extract` flag is actualy redundant because it is set by default (see the last sentence in the pasted excerpt from the documentation).

Reply:
Removing `--extract` produces errors. I'd rather leave that alone and nag bioinformaticians.